### ISSUE TITLE
CityIDRegistry should return all matching IDs and locations

### ISSUE THREAD

**Issue by csparpa** [csparpa/pyowm#153]
Currently, `CityIDRegistry` lookup functions

```python
registry.id_for('city_name')
registry.location_for('city_name')
```

always return the *first* matching ID or location.

Conversely, what happens when multiple matching cities are found?

Let's introduce two new functions

```python
def ids_for(city_name, country=None, matching='nocase'):
  ...

def locations_for(city_name, country=None, matching='nocase'):
  ...
```

returning:
  - an empty `list` if no matching cities are found
  - a `list` of tuples in the form `(id, city_name, country)` or a `list` of  `Location` objects upon 1+ matching cities found

Optional parameters:

 `country` --> specifies the country where to lookup for the city name. This restricts the search only to specific countries, given that usually towns with the same name exist in different countries (eg: London is both in the US and in the UK, and multiple Londons exist in the US). Defaults to: `None`, meaning: lookup into all countries

 `matching` --> specifies the type of string matching:
  - `exact` means: literal matching (case-sensitive). 
  - `nocase` means: case-insensitive matching
  - `like` means: containing a substring like the one provided, this can retrieve

The parameter value defaults to `nocase`

Example:
```python
results = reg.ids_for('London')    # [ (123, 'London', 'UK'), (456, 'London', 'MA'), (789, 'London', 'WY')]
results = reg.ids_for('London', country='UK')  # [ (123, 'London, UK') ]
results = reg.ids_for('Neverland')  # []
```

**Comment by danshorstein** [comment]
Is there any way to search for location by zip code? That might help with multiple cities with the same name.

**Comment by csparpa** [comment]
@danshorstein I'm afraid we don't have such an info in the source lists :-/

**Comment by csparpa** [comment]
Fixed with PR #157
